### ISSUE TITLE
Panic recursion detected when linking a scroll view's viewport-width to its child preferred with.

### ISSUE THREAD

**Issue by abique** [slint-ui/slint#4458]
To repro, checkout `120b980ee9c59e13e5ee868a106ef949e46364d5` in my repository, then `cargo run`.

The problem is in `ui/library_results.slint` line 175.

@ogoffart should you make this issue "private" as it is related to my repo which isn't open source?

**Comment by ogoffart** [comment]
Thanks for the bug report.

Here is a minimized testcase

``
import { ScrollView } from "std-widgets.slint";
export component TestCase {
    ScrollView {
        viewport-width <=> i-table.preferred-width;
        i-table := Rectangle {
            HorizontalLayout {
                for title in [""] : Rectangle {
                    width: i-table.width ;
                }
            }
        }
    }
}
``` 

Looks like the loop detection didn't detect this loop for some reason. Maybe related to the two way binding

**Comment by ogoffart** [comment]
Reproducer from #10343

```slint
component ScrollView {
    in-out property <length> viewport-height <=> flickable.viewport-height;
    flickable := Flickable {
        @children
    }
}

export component ScrollableContent inherits Window {
    ScrollView {
        viewport-height: rr.height;
        rr := Rectangle { }
    }
}
```


Even smaller testcase:
```slint
component Xxx {
    in-out property <length> prop <=> prop2;
    property <length> prop2;
    out property <length> prop3 <=> prop2;
}

export component Example inherits Window {
    Xxx {
        prop: self.prop3;
        height: self.prop;
    }
}
```
